src/product.h

```diff
@@ -77,6 +77,9 @@ class Product : public Resource {
   /// Changes the product's package id
   virtual void ChangePackageId(int new_package_id = Package::unpackaged_id());
 
+  // Packages the product
+  virtual std::vector<Product::Ptr> Package(Package::Ptr pkg); 
```

Comment:
This one also needs to be
```suggestion
  virtual std::vector<Resource::Ptr> Package(Package::Ptr pkg); 
```